### PR TITLE
Task#2 - Feat: Create button component 

### LINKED ISSUE #8: Task#2 - DevTask: Create button component
## OVERVIEW
- Create common components: Button
- Create props for button
- Testing

### PR DESCRIPTION
**Pull request by TheHienDuong**
- Closes #8 
## OVERVIEW
- Add new rule in eslint.cjs to fix linebreak-sytle error
- Clean up test code
- Create 3 button ( Login, Try for free, Try for free ( rightIcon ) ) 
- Clean up existing styles on first install
- Change the properties of color, font Weight, line Height for the tailwind library

### DEMO
image

### CODE REVIEW

File: practice-one/src/Components/Common/Buttons.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+function Button({
+  text, variant, size, rightIcon, type, color, onClick,
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/Buttons.jsx]: Please break line for each property. Update ALL similar places

File: practice-one/src/App.jsx [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+    console.log('Button clicked!');
+  };
+
   return (
-    <div className="App">
-      <h1 className="text-3xl text-purple-400 font-bold underline">
-        Hello world!
-      </h1>
+    <div>
+      <Button text="login" variant="primary" onClick={handleClick} />
```
Comment (phattruong-agilityio) [inline practice-one/src/App.jsx]: Please break line for each property. Update ALL similar places

File: practice-one/src/Components/Common/Buttons.jsx (hunk 1 above)
Comment (phattruong-agilityio) [reply practice-one/src/Components/Common/Buttons.jsx]: Missing default value

File: practice-one/src/Components/Common/Buttons.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+};
+
+const colorClasses = {
+  green: 'text-green-500',
+  blue: 'text-blue-500',
+  red: 'text-red-500',
+  gray: 'text-gray-500',
+};
+
+export default Button;
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/Buttons.jsx:59]: Please apply the proptypes

File: practice-one/src/index.css [hunk 4]
```diff
@@ -2,45 +2,6 @@
...
-  min-width: 320px;
-  min-height: 100vh;
-}
-
-h1 {
-  font-size: 3.2em;
-  line-height: 1.1;
-}
-
 button {
```
Comment (phattruong-agilityio) [inline practice-one/src/index.css]: we are using tailwind. Please make it correct

Comment (phattruong-agilityio) [reply practice-one/src/index.css]: Do we need this css style in tailwind

File: practice-one/src/Components/Common/Buttons.jsx (hunk 1 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/Buttons.jsx]: Done!

File: practice-one/src/App.jsx (hunk 2 above)
Comment (TheHienDuong) [reply practice-one/src/App.jsx]: Thanks! Done

File: practice-one/src/Components/Common/Buttons.jsx (hunk 3 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/Buttons.jsx:59]: Thanks! I updated and added proptyypes and default value for button

File: practice-one/src/index.css (hunk 4 above)
Comment (TheHienDuong) [reply practice-one/src/index.css]: Thanks! Removed style  into variantClass

File: practice-one/src/Components/Common/Buttons.jsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  type,
+  color,
+  onClick = () => {},
+}) => {
+  // Define the Tailwind CSS classes based on the props
+  const buttonClasses = `inline-flex items-center justify-center rounded ${sizeClasses[size]} ${variantClasses[variant]} ${colorClasses[color]}`;
+
+  return (
+    <button
+      type={type ? 'button' : 'submit'}
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/Buttons.jsx]: wrong type. is `type`  boolean ?

File: practice-one/src/Components/Common/Buttons.jsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  text: PropTypes.string.isRequired,
+  variant: PropTypes.oneOf(['primary', 'secondary', 'outline']),
+  size: PropTypes.oneOf(['small', 'medium', 'large']),
+  rightIcon: PropTypes.node,
+  type: PropTypes.oneOf(['button', 'submit', 'reset']),
+  color: PropTypes.oneOf(['green', 'blue', 'red', 'gray']),
+  onClick: PropTypes.func,
+};
+
+Button.defaultProps = {
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/Buttons.jsx]: It's totally OK but it's could be better if we do like this. more easily to read!

```js
const Button = ({
  text,
  variant = 'primary',
  size = 'medium',
  rightIcon,
  type = 'button',
  color = 'black',
  onClick = () => {},
}) => {
```

Comment (phattruong-agilityio) [reply practice-one/src/Components/Common/Buttons.jsx]: Do we still need this one?

File: practice-one/src/Components/Common/Buttons.jsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  type,
+  color,
+  onClick = () => {},
+}) => {
+  // Define the Tailwind CSS classes based on the props
+  const buttonClasses = `inline-flex items-center justify-center rounded ${sizeClasses[size]} ${variantClasses[variant]} ${colorClasses[color]}`;
+
+  return (
+    <button
+      type={Boolean ? 'button' : 'submit'}
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/Buttons.jsx]: what is this?
Did you understand the prop-type

```js
type: PropTypes.oneOf(['button', 'submit', 'reset']),
```

`type` is one of ['button', 'submit', 'reset'] why have "boolean" here

Comment (phattruong-agilityio) [reply practice-one/src/Components/Common/Buttons.jsx]: What is "Boolean"? does it always return true? where is the "reset" value?

File: practice-one/src/Components/Common/Buttons.jsx (hunk 6 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/Buttons.jsx]: Yes, I removed it

File: practice-one/src/Components/Common/Buttons.jsx (hunk 7 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/Buttons.jsx]: Done

**Review by phattruong-agilityio — APPROVED** [review]

**Pull request merged**